drivers/memc/memc_nxp_flexram.h

```diff
@@ -44,20 +44,27 @@ void memc_flexram_register_callback(flexram_callback_t callback, void *user_data
  * call from platform_init to set up flexram if using runtime map
  * must be inlined because cannot use stack
  */
-#define GPR17_REG_FILL(node_id, prop, idx) + (DT_PROP_BY_IDX(node_id, prop, idx) << (2*idx))
+#define GPR_FLEXRAM_REG_FILL(node_id, prop, idx) \
+	(((uint32_t)DT_PROP_BY_IDX(node_id, prop, idx)) << (2 * idx))
 static inline void memc_flexram_dt_partition(void)
 {
 	/* iomuxc_gpr must be const (in ROM region) because used in reconfiguring ram */
 	static IOMUXC_GPR_Type *const iomuxc_gpr =
-		(IOMUXC_GPR_Type *) DT_REG_ADDR(IOMUXC_GPR_DT_NODE);
+		(IOMUXC_GPR_Type *)DT_REG_ADDR(IOMUXC_GPR_DT_NODE);
 	/* do not create stack variables or use any data from ram in this function */
-	iomuxc_gpr->GPR17 = DT_FOREACH_PROP_ELEM_SEP(FLEXRAM_DT_NODE,
-				flexram_bank_spec, GPR17_REG_FILL, (+));
+#if defined(CONFIG_SOC_SERIES_IMXRT11XX)
```

Comment:
Rather than a SoC define, is there a feature file macro that can be should? Or should we add a Kconfig to identify this?

Reply:
if in the future there are more platforms with flexram then I guess we can consider making it more complicated, right now I'm not sure what the best approach would be, this is fair enough I think since only the RT10 and RT11 afaik have this flexram